.github/workflows/auto_roll.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+set -ex
+
+browser_name="$1"
+playwright_dir="$(pwd)"
+
+function set_github_action_env {
+    echo "::set-env name=$1::$2"
+}
+
+git config --global user.email "microsoftopensource@users.noreply.github.com"
+git config --global user.name "Microsoft Open Source"
+export SHELL="/bin/bash"
+
+# Ensure the compiled browser will be used
+rm -rf ~/.cache/ms-playwright/
+
+if [ "$browser_name" = "chromium" ]; then
+    ./browser_patches/chromium/build.sh --linux
+    set_github_action_env "CRPATH" "$(pwd)/browser_patches/chromium/output/chrome-linux/chrome"
+    exit 0
+elif [ "$browser_name" = "webkit" ]; then
+    sudo apt install -y libharfbuzz-dev libepoxy-dev libgcrypt-dev libsoup2.4-dev libwebp-dev flatpak
+elif [ "$browser_name" = "firefox" ]; then
+    sudo apt install -y autoconf2.13 libclang-dev clang
+fi
+
+./browser_patches/prepare_checkout.sh "$browser_name"
+
```

Comment:
Right here we can rebaseline our browser and even create a new commit to the repo if needed. This is instead of the `utils/roll_browser_tip_of_tree.js` script you had to write manually:

```sh
if [[ "${browser_name}" == "webkit" ]]; then
  cd ./browser_patches/webkit/checkout
  # Rebase WebKit atop of master branch.
  git rebase browser_upstream/master
  cd -
elif [[ "${browser_name}" == "firefox" ]]; then
  cd ./browser_patches/firefox/checkout
  # We keep firefox atop of beta branch since it's much more stable.
  git rebase browser_upstream/beta
  cd -
fi

# export changes
./browser_patches/export.sh "${browser_name}"
git commit -am "rebaseline browser!"
```

Reply:
thats great!